chat-plugins/blackjack.js

```diff
@@ -0,0 +1,604 @@
+		this.players[user].selfUhtml = 'change';
+		this.players[user].resetTimerTicks();
+
+		this.giveCard(user);
+	}
+	stand(user) {
+		const player = this.players[user];
+		if (this.state !== 'started') return this.errorMessage(user, `Blackjack hasn't started yet.`);
+		if (!player) return this.errorMessage(user, `You aren't a player in this game.`);
+		if (this.curUser !== user.userid) return this.errorMessage(user, `It's not your turn.`);
+		player.status = 'stand';
+		let cards = '';
+		for (let u in player.cards) cards += `[${player.cards[u]}] `;
+		this.display(Chat.html`<br /><strong>${player.name}</strong> <u>stands</u> with ${cards} (${player.points})`, null, player.name);
+		this.clear();
+
+		this.next();
+	}
+	giveCard(user) {
+		if (this.deck.length < 1) this.deck = Dex.shuffle(this.deck);
+		const player = (user === 'dealer' ? this.dealer : this.players[user]);
+		if (!player) return; // this should never happen
+		player.cards.push(this.deck[0]);
+
+		this.deck.shift();
+
+		if (this.deck.length === 0) {
+			this.display(`<br />${this.dealer.name} has ran out of cards in the deck; shuffling a new deck...`);
+			this.deck = new BlackjackDeck().shuffle();
+		}
+
+		player.points = this.getCardPoints(player);
+
+		if (player.cards.length < 3) return;
+
+		if (player.cards.length > 2) this.display(Chat.html`<br /><strong>${player.name}</strong> <u>hit</u> and received [${player.cards[player.cards.length - 1]}] (${player.points})`, null, player.name);
+
+		if (user === 'dealer') {
+			if (player.points > 21) {
+				let cards = '';
+				for (let u in player.cards) cards += `[${player.cards[u]}] `;
+				this.display(Chat.html`<br /><strong>${this.dealer.name}</strong> has <u>busted</u> with ${cards} (${player.points})`);
+				this.end();
+				return;
+			} else if (player.points === 21) {
+				this.display(`<br /><strong>${this.dealer.name}</strong> has blackjack!`);
+				this.end();
+				return;
+			}
+		}
+
+		if (player.points > 21) {
+			player.status = 'bust';
+			let cards = '';
+			for (let u in player.cards) cards += `[${player.cards[u]}] `;
+			this.display(Chat.html`<br /><strong>${player.name}</strong> has <u>busted</u> with ${cards} (${player.points})`, null, player.name);
+			this.clear();
+		}
+		if (player.points === 21) {
+			player.status = 'stand';
+			this.display(Chat.html`<br /><strong>${player.name}</strong> has blackjack!`, null, player.name);
+			this.clear();
+		}
+		if (user !== 'dealer') this.players[user].cards = player.cards;
+		this.next();
+	}
+	getCardPoints(player) {
+		let points = 0;
+		let aceCount = 0;
+		for (let i in player.cards) {
+			let card = toId(player.cards[i]).toUpperCase();
+			if (!isNaN(Number(card))) {
+				points += Number(card);
+			} else if (['K', 'Q', 'J'].includes(card)) {
+				points += 10;
+			} else if (card === 'A') {
+				points += 11;
+				aceCount++;
+			}
+		}
+
+		// At first, we value aces as 11, however, we will change their value
```

Comment:
Is there a way we could make a popup occur on getting the Ace card saying "Choose a value: 11, 1" since it could be beneficial to manually choose that to possibly get a Blackjack, just an idea